paddle/fluid/pir/serialize_deserialize/include/serialize_utils.h

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) 2024 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef PADDLE_FLUID_PIR_SERIALIZE_DESERIALIZE_INCLUDE_SERIALIZE_UTILS_H_
+#define PADDLE_FLUID_PIR_SERIALIZE_DESERIALIZE_INCLUDE_SERIALIZE_UTILS_H_
+#include <fstream>
+#include <initializer_list>
+#include <string>
+#include <vector>
+
+#include "glog/logging.h"
+#include "nlohmann/json.hpp"
+#include "paddle/fluid/pir/dialect/operator/ir/op_attribute.h"
+#include "paddle/fluid/pir/serialize_deserialize/include/utils.h"
+#include "paddle/phi/common/data_type.h"
+#include "paddle/pir/include/core/builtin_attribute.h"
+#include "paddle/pir/include/core/builtin_type.h"
+using Json = nlohmann::json;
+namespace pir {
+
+template <typename T>
+Json serializeTypeToJson(const T& type) {
+  Json j;
```

Comment:
```suggestion
  Json json;
```
i,j,k 这些一般默认是index类，不建议这样缩写

Reply:
done